containers/java-postgres/test-project/test-utils.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/bash
+SCRIPT_FOLDER="$(cd "$(dirname $0)" && pwd)"
+USERNAME=${1:-vscode}
+
+if [ -z $HOME ]; then
+    HOME="/root"
+fi
+
+FAILED=()
+
+echoStderr()
+{
+    echo "$@" 1>&2
+}
+
+check() {
+    LABEL=$1
+    shift
+    echo -e "\n🧪 Testing $LABEL"
+    if "$@"; then 
+        echo "✅  Passed!"
+        return 0
+    else
+        echoStderr "❌ $LABEL check failed."
+        FAILED+=("$LABEL")
+        return 1
+    fi
+}
+
+checkMultiple() {
+    PASSED=0
+    LABEL="$1"
+    echo -e "\n🧪 Testing $LABEL."
+    shift; MINIMUMPASSED=$1
+    shift; EXPRESSION="$1"
+    while [ "$EXPRESSION" != "" ]; do
+        if $EXPRESSION; then ((PASSED++)); fi
+        shift; EXPRESSION=$1
+    done
+    if [ $PASSED -ge $MINIMUMPASSED ]; then
+        echo "✅ Passed!"
+        return 0
+    else
+        echoStderr "❌ $LABEL check failed."
+        FAILED+=("$LABEL")
+        return 1
+    fi
+}
+
+checkOSPackages() {
+    LABEL=$1
+    shift
+    echo -e "\n🧪 Testing $LABEL"
+    if dpkg-query --show -f='${Package}: ${Version}\n' "$@"; then 
+        echo "✅  Passed!"
+        return 0
+    else
+        echoStderr "❌ $LABEL check failed."
+        FAILED+=("$LABEL")
+        return 1
+    fi
+}
+
+checkExtension() {
+    # Happens asynchronusly, so keep retrying 10 times with an increasing delay
+    EXTN_ID="$1"
+    TIMEOUT_SECONDS="${2:-10}"
+    RETRY_COUNT=0
+    echo -e -n "\n🧪 Looking for extension $1 for maximum of ${TIMEOUT_SECONDS}s"
+    until [ "${RETRY_COUNT}" -eq "${TIMEOUT_SECONDS}" ] || \
+        [ ! -e $HOME/.vscode-server/extensions/${EXTN_ID}* ] || \
+        [ ! -e $HOME/.vscode-server-insiders/extensions/${EXTN_ID}* ] || \
+        [ ! -e $HOME/.vscode-test-server/extensions/${EXTN_ID}* ] || \
+        [ ! -e $HOME/.vscode-remote/extensions/${EXTN_ID}* ]
+    do
+        sleep 1s
+        (( RETRY_COUNT++ ))
+        echo -n "."
+    done
+
+    if [ ${RETRY_COUNT} -lt ${TIMEOUT_SECONDS} ]; then
+        echo -e "\n✅ Passed!"
+        return 0
+    else
+        echoStderr -e "\n❌ Extension $EXTN_ID not found."
+        FAILED+=("$LABEL")
+        return 1
+    fi
+}
+
+checkCommon()
+{
```

Comment:
Same comment as above here as well

Reply:
`test-utils.sh` is a 'shared' file that's the same in all containers, so we've not made any edits to it